### PR TITLE
Managing DNS zones with terraform

### PR DESCRIPTION
**Pull request by RMcVelia**
Add terraform code to manage Azure DNS zones and records

Add configuration to create new register zone

https://trello.com/c/5p8AnNqF/555-create-dns-zone-register-trainee-teachersservicegovuk

### CODE REVIEW

File: dns/zones/terraform.tf [hunk 1]
```diff
@@ -0,0 +1,22 @@
+terraform {
+  required_version = "~> 0.15.5"
```
Comment (saliceti) [inline dns/zones/terraform.tf:2]: Can we update terraform and the azurerm module to the latest versions?

File: dns/records/workspace-variables/pubtest_staging.tfvars.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        },
+        "staging-val": {
+          "target": "d3kffbwt0testc.cloudfront.net",
+          "ttl": 300
+        }
+      },
+    "resource_group_name": "s121d01-dns"
+    }
+  }
+}
```
Comment (saliceti) [inline dns/records/workspace-variables/pubtest_staging.tfvars.json:30]: Missing line at end of file. I suggest setting "Insert Final Newline" and "Trim Final Newlines" in vscode to do it automatically

File: dns/zones/terraform.tf [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "2.62.0"
+    }
+  }
+  backend "azurerm" {
+  }
+  experiments = [module_variable_optional_attrs]
```
Comment (saliceti) [inline dns/zones/terraform.tf:11]: I don't think this is required (also for dns/records)

File: dns/zones/terraform.tf (hunk 1 above)
Comment (RMcVelia) [reply dns/zones/terraform.tf]: updated tf and azurerm to latest versions

File: dns/records/workspace-variables/pubtest_staging.tfvars.json (hunk 2 above)
Comment (RMcVelia) [reply dns/records/workspace-variables/pubtest_staging.tfvars.json]: done

File: dns/zones/terraform.tf (hunk 3 above)
Comment (RMcVelia) [reply dns/zones/terraform.tf]: removed experiments (and retested without)

**Review by saliceti — APPROVED** [review]

**Pull request merged**